patroni/ha.py

```diff
@@ -193,12 +193,18 @@ def is_standby_cluster(self) -> bool:
         return self.global_config.is_standby_cluster
 
     def is_leader(self) -> bool:
-        with self._is_leader_lock:
-            return self._is_leader > time.time()
+        """:returns: `True` if the current node is the leader, based on expiration set when it last held the key."""
+        with self._leader_expiry_lock:
+            return self._leader_expiry > time.time()
 
     def set_is_leader(self, value: bool) -> None:
-        with self._is_leader_lock:
-            self._is_leader = time.time() + self.dcs.ttl if value else 0
+        """Updates the current node's view of it's own leadership status. Will update the expiry timestamp to match
```

Comment:
http://www.pydocstyle.org/en/2.1.1/error_codes.html

D205, D209, D400 (you can use flake8-docstrings to track if all issues are resolved)